services/continuous_querier/service.go

```diff
@@ -330,6 +331,7 @@ func (s *Service) runContinuousQueryAndWriteResult(cq *ContinuousQuery) error {
 	if res.Err != nil {
 		return res.Err
 	}
+	close(closing)
```

Comment:
Can you move the `close(closing)` right after the creation (L320) and do it as a defer. That way it'll handle it for panics too. If it closes and the results are done reading the it shouldn't do anything anyway.